pkg/controller/dnszone/awsactuator.go

```diff
@@ -433,22 +434,39 @@ func (a *AWSActuator) Delete() error {
 
 func (a *AWSActuator) deleteRecordSets(logger log.FieldLogger) error {
 	logger.Info("Deleting route53 recordsets in hostedzone")
+
+	return DeleteAWSRecordSets(a.awsClient, a.dnsZone, []string{}, logger)
+}
+
+// DeleteAWSRecordSets will clean up all non-NS and non-SOA record types from a DNS zone when no
+// deleteRecordTypes is provided.
+// When providing a list of record types via deleteRecordTypes, then only those matching records will be deleted.
+func DeleteAWSRecordSets(awsClient awsclient.Client, dnsZone *hivev1.DNSZone, deleteRecordTypes []string, logger log.FieldLogger) error {
+
+	deleteSet := sets.NewString(deleteRecordTypes...)
 	maxItems := "100"
 	listInput := &route53.ListResourceRecordSetsInput{
-		HostedZoneId: a.zoneID,
+		HostedZoneId: dnsZone.Status.AWS.ZoneID,
 		MaxItems:     &maxItems,
 	}
 	for {
-		listOutput, err := a.awsClient.ListResourceRecordSets(listInput)
+		listOutput, err := awsClient.ListResourceRecordSets(listInput)
 		if err != nil {
 			return err
 		}
 		var changes []*route53.Change
 		for _, recordSet := range listOutput.ResourceRecordSets {
 			// Ignore the 2 recordsets that are created with the hosted zone and that cannot be deleted
-			if n, t := aws.StringValue(recordSet.Name), aws.StringValue(recordSet.Type); n == controllerutils.Dotted(a.dnsZone.Spec.Zone) && (t == route53.RRTypeNs || t == route53.RRTypeSoa) {
+			if n, t := aws.StringValue(recordSet.Name), aws.StringValue(recordSet.Type); n == controllerutils.Dotted(dnsZone.Spec.Zone) && (t == route53.RRTypeNs || t == route53.RRTypeSoa) {
 				continue
 			}
+
+			// If we've been provided a list of record set types to delete
+			// and the current recordSet is not in that list, skip deleting it.
+			if deleteSet.Len() > 0 && !deleteSet.Has(*recordSet.Type) {
```

Comment:
I understand that you are trying to preserve the existing functionality, but I don't think this is necessary. There is no harm in deleting all of the recordsets in between installs the same as we do for a deprovision.